lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4SharedOnChipRam.scala

```diff
@@ -60,60 +60,123 @@ case class Axi4SharedOnChipRam(dataWidth : Int, byteCount : BigInt, idWidth : In
 }
 
 
-//WARNING, do not support backpresure
-case class Axi4SharedOnChipRamMultiPort(portCount : Int, dataWidth : Int,byteCount : BigInt,idWidth : Int) extends Component{
-  val axiConfig = Axi4SharedOnChipRam.getAxiConfig(dataWidth,byteCount,idWidth)
+case class Axi4SharedOnChipRamPort(config: Axi4Config) extends Bundle {
+    def attach(ram: Mem[Bits]): Axi4Shared = {
+        val wordRange       = config.wordRange
+        val axi             = Axi4Shared(config)
+        val arw             = axi.arw.unburstify
+        val writeDataStream = axi.writeData
+
+        val readStream      = Stream(Axi4R(axi.config))
+        val readAddrStream  = cloneOf(arw)
+        val writeStream     = Stream(Axi4B(axi.config))
+        val writeAddrStream = cloneOf(arw)
+
+        Vec(readAddrStream, writeAddrStream) <> StreamDemux(
+            arw,
+            U(arw.write),
+            2
+        )
+
+        val writeCombined = StreamJoin(writeAddrStream, writeDataStream)
+        val writeAddr     = writeCombined._1.addr(wordRange)
+        ram.write(
+            address = writeAddr.resized,
+            data = writeCombined._2.data,
+            enable = writeCombined.fire,
+            mask = writeCombined._2.strb
+        )
+
+        writeStream.translateFrom(writeCombined) { (to, from) =>
+            if (axi.config.useId) to.id := from._1.id
+            if (axi.config.useWUser) to.user := from._1.user
+            to.setOKAY()
+        }
+        val writeReady = RegInit(False).riseWhen(True)
+        writeStream.ready := writeReady
+
+        val bStream      = cloneOf(writeStream)
+        val writeLast    = writeCombined.fire && writeCombined._1.last
+        val writePayload = RegNextWhen(writeStream.payload, writeLast)
+        val writeDone    = RegInit(False)
+        when(writeLast) {
+            writeDone := True
+        } elsewhen (bStream.fire) {
+            writeDone := False
+        }
+        bStream.valid := writeDone
+        bStream.payload := writePayload
+        axi.writeRsp << bStream
+
+        val readAddr = readAddrStream.addr(wordRange)
+        val readData = ram
+            .readSync(
+                address = readAddr.resized
```

Comment:
I'm a bit worrying about the inferation, basicaly, the behind Axi4SharedOnChipRamPort, was to use a true-dual-port ram to infer a two port axi ram.

Here, the address ued for the readSync and the write aren't the same signal, this may confuse synthesis ?

Reply:
Does that means readSync and write should use the same address signal to help synthesizer correctly infer a true-dual-port?

Reply:
I tried this in Vivado, there is no trouble to it's synthesizer which generate a correct netlist. A real dual block ram has been initiated.

Reply:
Issue is, vivado may be a bit less stupid than other synthesis tools, but in general, it is a realy pain in the ass.

Reply:
ahh，I will use a standalone addr to do that